Modify respond_to behaviour always setting the response's content type based on the request format (#37617)
rails/rails: actionpack/CHANGELOG.md

*   `respond_to#any` no longer returns a response's Content-Type based on the
    request format but based on the block given.

    Example:

    ```ruby
      def my_action
        respond_to do |format|
          format.any { render(json: { foo: 'bar' }) }
        end
      end

      get('my_action.csv')
    ```

>     Previous behaviour was returning a `text/csv` response's Content-Type which

Reviewer: ```suggestion
    The previous behaviour was to respond with a `text/csv` Content-Type which
```

>     is inaccurate since we are returning a JSON response.

Reviewer 2: ```suggestion
    is inaccurate since a JSON response is being rendered.
```

>     is innacurate since we are returning a JSON response.

Reviewer: ```suggestion
    is inaccurate since we are returning a JSON response.
```

>     Current behaviour correctly returns a `application/json` response's Content-Type

Reviewer: ```suggestion
    Now it correctly returns a `application/json` Content-Type.
```
